### PR TITLE
fix: enable auto-compaction of etcd by default

### PR DESCRIPTION
**Pull request by moshloop**
Enable auto-compaction of etcd to prevent mvcc: database space exceeded errors

```release-note
cilium-etcd-operator: enable auto-compaction by default
```

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/7399)

### CODE REVIEW

**Review by aanm — CHANGES_REQUESTED** [review]
@moshloop thanks for the change. Can you provide a reason why `1` is a good compaction retention?

**Comment by moshloop** [comment]
I am not sure if it is necessary to retain any versions and 1 (or 1h) seems to be a common value

**Comment by aanm** [comment]
> I am not sure if it is necessary to retain any versions and 1 (or 1h) seems to be a common value

@moshloop Ah ok. That makes more sense. We have been testing with "25000", not hours but based on the revision number. Would that sound sane to you as well?

**Comment by moshloop** [comment]
Calico uses a [1h](https://github.com/projectcalico/calico/blob/master/master/getting-started/kubernetes/installation/hosted/etcd.yaml#L58) window.  Seeing as etcd is going to run a compaction job periodically anyway and I don't think there really is a use case to go back in time (as I understand etcd is used more as a cache as it's populated on creation anyway) it is probably safer to have predictable compaction rather than risk storage being exceeded.

**Comment by aanm** [comment]
> Calico uses a [1h](https://github.com/projectcalico/calico/blob/master/master/getting-started/kubernetes/installation/hosted/etcd.yaml#L58) window. Seeing as etcd is going to run a compaction job periodically anyway and I don't think there really is a use case to go back in time (as I understand etcd is used more as a cache as it's populated on creation anyway) it is probably safer to have predictable compaction rather than risk storage being exceeded.

@moshloop agreed. Can you then change the file in `examples/kubernetes/templates/v1/cilium-etcd-operator.yaml.sed` by adding this?
```
        - name: CILIUM_ETCD_META_ETCD_AUTO_COMPACTION_MODE
          value: "revision"
        - name: CILIUM_ETCD_META_ETCD_AUTO_COMPACTION_RETENTION
          value: "25000"
```

The file that you changed is auto generated so you will need to run `CILIUM_VERSION=latest make -C examples/kubernetes clean all` and commit all files modified

**Comment by moshloop** [comment]
Would a revision based compaction strategy still not potentially lead to out of space if there is lots of churn, but not all on the same key?

**Comment by aanm** [comment]
> Would a revision based compaction strategy still not potentially lead to out of space if there is lots of churn, but not all on the same key?

@moshloop Can you elaborate? If I understand the etcd behavior correctly is that having the compaction on a time basis can have more probability to run out of space since in between that time period lots of things can happen.

**Comment by moshloop** [comment]
My bad - I was under the mistaken impression that revisions are per key, and not global in which case I agree revision based compaction is more predictable

File: examples/kubernetes/README.rst [hunk 1]
```diff
@@ -25,6 +25,8 @@ the directory :code:`templates`. The components ending with :code:`.sed` will be
 automatically generated based on the template itself and the specific
 :code:`transforms2sed.sed` inside each directory for each Kubernetes version.
 
+To make a change to a template and all versions of Kubernetes run: :code:`CILIUM_VERSION=latest make -C examples/kubernetes clean all`
```
Comment (aanm) [inline examples/kubernetes/README.rst:28]: Thanks for adding this!

**Review by aanm — APPROVED** [review]

**Comment by aanm** [comment]
test-me-please

**Review by nebril — APPROVED** [review]

**Comment by aanm** [comment]
test-me-please

**Pull request merged**